.gitignore

```diff
@@ -5,53 +5,73 @@
 # created during manual debugging and many people like to clean up instead of
 # having git ignore such leftovers. You can use `.git/info/exclude` to
 # configure your local ignore list.
-# FIXME: This needs cleanup.
-*~
+
+# File system
+.DS_Store
+desktop.ini
+
+# Version control
+.hgignore
+.hg/
+
+# Editor
 *.swp
 *.swo
-.#*
-.DS_Store
+Session.vim
 .cproject
-.hg/
-.hgignore
 .idea
 *.iml
-__pycache__/
-*.py[cod]
-*$py.class
+.vscode
 .project
+.favorites.json
 .settings/
+
+# Tool
 .valgrindrc
-.vscode
-.favorites.json
-/Makefile
-/build/
+.cargo
+# Included because it is part of the test case
+!/src/test/run-make/thumb-none-qemu/example/.cargo
+
+# Configuration
 /config.toml
-/dist/
+/Makefile
+config.mk
+config.stamp
+no_llvm_build
+
+# Build
 /dl/
 /doc/
 /inst/
 /llvm/
 /mingw-build/
-/src/tools/x/target
-# Created by default with `src/ci/docker/run.sh`:
```

Comment:
Why this comment is removed?

Reply:
I thought those comments applied to all the entries below them, and that categorizing by use case would be more clear, since something like `/target` could be generated by more than just `run.sh`. However, I checked the history and the comments only apply to the `/tmp/` and `/obj/` directory, which is why I now added them again. I changed the formatting a bit to make it clearer they are documenting only 1 entry.